lineman/app/components/navbar/navbar.coffee

```diff
@@ -3,5 +3,13 @@ angular.module('loomioApp').directive 'navbar', ->
   restrict: 'E'
   templateUrl: 'generated/components/navbar/navbar.html'
   replace: true
-  controller: 'NavbarController'
-  link: (scope, element, attrs) ->
+  controller: ($scope, Records, ThreadQueryService) ->
+    $scope.$on 'currentComponent', (el, component) ->
+      $scope.selected = component
+
+    $scope.unreadThreadCount = ->
+      ThreadQueryService.unreadQuery().length()
+
+    if !$scope.inboxLoaded
+      Records.discussions.fetchInbox().then ->
```

Comment:
Make sure we're just loading the inbox once, then later we'll do time-sensitive queries for (what should be) tiny live updates to the inbox.